src/Runner.php

```diff
@@ -146,18 +155,17 @@ public function start() {
 		Logger::success( 'Created pull request with composer.lock changes.' );
 	}
 
-	/**
-	 * Check that Git, Composer, and Hub are available on the filesystem.
-	 */
-	private function check_executables() {
-		$execs = array( 'git', 'composer', 'hub' );
-		foreach( $execs as $exec ) {
-			exec( 'type ' . escapeshellarg( $exec ), $_, $return_code );
-			if ( 0 !== $return_code ) {
-				Logger::error( "Missing {$exec} on the system." );
+	private function checkExisting() {
+		exec('hub issue', $output_lines, $return_code);
+		if ( 0 !== $return_code ) {
+			Logger::error( 'Unable to check for existing pull requests with hub.' );
+		}
+		foreach ($output_lines as $line) {
+			if (preg_match('%Update Composer dependencies \(([0-9-]*)\)%', $line, $matches)) {
+				// We will presume the branch name is 'clu-' followed by the date in the issue title.
+				return 'clu-' . $matches[1];
 			}
 		}
-		Logger::info( 'Found required executables on system: ' . implode( ', ', $execs ) );
+		return false;
 	}
-
 }
```

Comment:
Any ideas on how we could get some basic unit or integration tests going for this?

Reply:
Functional tests for the basic functionality (pre this PR), it would be pretty easy. We could use `composer create-project` to clone a project that has out-of-date components, and then run `clu` on it. The [Terminus Build Tools plugin](https://github.com/pantheon-systems/terminus-build-tools-plugin) uses similar techniques in its tests.

Functional tests for re-using a PR would be more difficult. Maybe we could cheat a bit to do this step. Create a PR per the above with clu; then, edit the `composer.json` to add a component that was not previously included in the system-under-test. At that point, running `clu` again will behave the same way that it would when there was an update. So, while we would not be testing the same use-case, we would be exercising the code in the same way, and I think that would be sufficient. Creating a mock packagist.org would be too much trouble, and mocking the API calls with something like VCR would also be more work, and less valuable.

I don't think that unit tests for this project would be particularly valuable.

Maybe we could tackle testing as a follow-on task? I have limited time prior to DrupalCon Vienna, but would be happy to help out with this on my return.